lib/runner_test.go

```diff
@@ -20,26 +21,35 @@ var (
 	redBoldString = color.New(color.FgRed, color.Bold).SprintfFunc()
 )
 
-func setupTestUI() (*bytes.Buffer, *bytes.Buffer) {
-	in, out := &bytes.Buffer{}, &bytes.Buffer{}
-	UI = termui.New(in, out, nil)
-	color.Output = UI
-	color.NoColor = false
-	return in, out
-}
-
-func setupDefaultRunner() *Runner {
-	return &Runner{
-		TestTargets: []string{},
-		IncludeRe:   defaultInclude,
-		ExcludeRe:   defaultExclude,
-		Timeout:     defaultTimeout,
-		InOrder:     false,
-		RandomSeed:  defaultSeed,
-		JSONOutput:  false,
-		Verbose:     false,
-		DryRun:      false,
-	}
+func setupDefaultRunner() (*Runner, io.ReadWriter, io.ReadWriter) {
+	testTargets := []string{}
+	includeReStr := defaultInclude
+	excludeReStr := defaultExclude
+	timeout := defaultTimeout
+	inOrder := false
+	randomSeed := defaultSeed
+	jsonOutput := false
+	verbose := false
+	dryRun := false
+	options := NewRunnerOptions(
```

Comment:
Not quiet what I meant by optional parameters since all the args just moved from one constructor to another, basically I was suggesting to initialize a struct that is called options which you don't necessarily need constructor for it. You can simply initialize options directly e.g 
```
testOptions := options{
etc
...
}
```

Reply:
Oh, I see.